akka-actor-typed/src/main/scala/akka/actor/typed/internal/StashBufferImpl.scala

```diff
@@ -125,20 +125,47 @@ import akka.util.ConstantFun
       val b2 = Behavior.start(b, ctx)
       if (!Behavior.isAlive(b2) || !messages.hasNext) b2
       else {
-        val nextB = try {
-          messages.next() match {
+        val message = messages.next()
+        val interpretResult = try {
+          message match {
             case sig: Signal => Behavior.interpretSignal(b2, ctx, sig)
             case msg         => Behavior.interpretMessage(b2, ctx, msg)
           }
         } catch {
           case NonFatal(e) => throw UnstashException(e, b2)
         }
 
-        interpretOne(Behavior.canonicalize(nextB, b2, ctx)) // recursive
+        val actualNext =
+          if (interpretResult == Behavior.same) b2
+          else if (Behavior.isUnhandled(interpretResult)) {
+            ctx.asScala.asInstanceOf[ActorContextAdapter[_]].onUnhandled(message)
+            b2
+          } else {
+            interpretResult
+          }
+
+        if (Behavior.isAlive(actualNext))
+          interpretOne(Behavior.canonicalize(actualNext, b2, ctx)) // recursive
+        else
+          // fixme rest of stash to dead letter?
```

Comment:
yes, I think that is the right thing to do

Reply:
Create issue so it's not lost in the depths and mark #number? Unless it's happening in this PR eventually of course.